File: benchmarks/pandas/bench_dataframe_filter.py
"""Benchmark: DataFrame filter (boolean mask on 100k-row DataFrame)"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 5
ITERATIONS = 20

vals = np.arange(ROWS, dtype=np.float64) * 0.1
df = pd.DataFrame({"value": vals})

for _ in range(WARMUP):
    df[df["value"] > 5000]

start = time.perf_counter()
for _ in range(ITERATIONS):
    df[df["value"] > 5000]
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_filter",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_filter.ts
/**
 * Benchmark: DataFrame filter (boolean mask on 100k-row DataFrame)
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 5;
const ITERATIONS = 20;

const vals = Array.from({ length: ROWS }, (_, i) => i * 0.1);
const df = DataFrame.fromColumns({ value: vals });
const valueSeries = df.col("value");

for (let i = 0; i < WARMUP; i++) {
  df.filter(valueSeries.gt(5000));
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  df.filter(valueSeries.gt(5000));
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_filter",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
